pkg/server/sandbox_status.go

```diff
@@ -58,24 +58,50 @@ func (c *criContainerdService) PodSandboxStatus(ctx context.Context, r *runtime.
 			state = runtime.PodSandboxState_SANDBOX_READY
 		}
 	}
+
+	ip, err := c.getIP(sandbox)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get sandbox ip: %v", err)
+	}
+
+	createdAt := sandbox.Container.Info().CreatedAt
+	status := toCRISandboxStatus(sandbox.Metadata, state, createdAt, ip)
+	return &runtime.PodSandboxStatusResponse{Status: status}, nil
+}
+
+func (c *criContainerdService) getIP(sandbox sandboxstore.Sandbox) (string, error) {
 	config := sandbox.Config
+
+	if config.GetLinux().GetSecurityContext().GetNamespaceOptions().GetHostNetwork() {
+		// For sandboxes using the host network we are not
+		// responsible for reporting the IP.
+		glog.V(10).Infof("getIP: %s: sandbox is using host network", sandbox.ID)
+		return "", nil
+	}
+
+	if err := c.netPlugin.Status(); err != nil {
+		// If the network is not ready then there is nothing to report.
+		glog.V(10).Infof("getIP: %s: network is not ready", sandbox.ID)
```

Comment:
This should not usually happen, because we should not be running pod which requires pod network when network plugin is not ready.

I think a V(4) log is ok here.

nit: `glog.V(4).Infof("getIP: network %q is not ready", sandbox.ID)`